### ISSUE TITLE
Server List Adjustments

### ISSUE THREAD

**Issue by cclough715** [jamulussoftware/jamulus#851]
Hello, 

With the ever growing list of public servers I think it would be helpful to implement a few new features to spice up the server list within the Jamulus client. Mostly these adjustments would be focused on improving accessibility to servers we have previously connected to. 

1. Implement a favorites feature so users can mark a server as a favorite. Then a user could select favorites from the list dropdown to filter all favorite servers. When a server is marked as a favorite there should be some indication in the server list such as an additional column with a star to indicate a favorite or a different colored name

2. If a user has marked as favorite then the server name should stay in the favorites list and if the server isn't running then display "N/A" for ping time to indicate the server is down or not running. This would help prevent the user spending time looking for something that isn't there. Alternatively, might be helpful to troubleshoot connection issues

3. Add the ability to save a private server to the list. If this feature would be implemented then the server client should be able to fill in the "My Server Info" table regardless of whether or not the server is public. That way the server can easily be identified. Additionally the client would be able to see the ping time and number of musicians before joining the server. And server hosts could easily save and connect to the localhost

Many thanks to all contributors!

**Comment by dcorson-ticino-com** [comment]
I have started to look at making this work.  I have added a tab in the connections window so it has either the Central Servers as 'til now or a list of favorites, sorted according to usage with the last used at the top.  A button on the clientdlg page adds the server currently in use to the favorite list. That way any server, public or private is treated the same.
My goal is to make this work even if the Central Servers are down, communication directly with the servers.  OK to get ping times and number of active clients, but I don't see a way to get the max clients from the servers. Am I missing something?
Don
